---
title: "RDA_dom_syl"
author: "Enrique Hernandez advised by Arne Jacobs"
date: "19/07/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(vegan)
library(tidyverse)
library(car)
library(geomorph)
library(gridExtra)
library(adegenet)
library(randomForest)
library(cowplot)

```


```{r}
geno <- read_delim("C:/Users/quiqu/Dropbox/Recuadoriensis_Analysis/RDA_analysis/RD_V2_analysis/KNNimpR3.domsylV2.genomat.012", delim = "\t", col_names = F)
phen <- read_csv("C:/Users/quiqu/Dropbox/Recuadoriensis_Analysis/RDA_analysis/RD_V2_analysis/domsyl.ipca.csv")


#Create matrix with all loci (select from first column with SNP info to last one in file)
gen <- geno[,2:2553]
gen <- as.matrix(gen)

#Impute missing data using na.roughfix function
gen.imputed <- na.roughfix(gen)
gen.imputed.dataset <- as_tibble(cbind(phen, gen.imputed))
genotypes <- gen.imputed.dataset[,6:2557]

test.rda <- rda(genotypes ~ Habitat + Condition(Axis1 + Axis2 + Axis3), data=gen.imputed.dataset, scale=T)

#                Inertia Proportion Rank
#Total         2.389e+03  1.000e+00     
#Conditional   5.985e+02  2.505e-01    3
#Constrained   2.737e+01  1.146e-02    1
#Unconstrained 1.763e+03  7.380e-01  103
#Inertia is correlations 

#Eigenvalues for constrained axes:
#  RDA1 
#27.371 

#Eigenvalues for unconstrained axes:
#  PC1   PC2   PC3   PC4   PC5   PC6   PC7   PC8 
#83.38 65.52 57.31 52.55 48.53 46.29 41.37 40.34 
#(Showing 8 of 103 unconstrained eigenvalues)


# Testing significance of predictors:
RsquareAdj(test.rda)
#$r.squared
#[1] 0.01145694
#$adj.r.squared
#[1] 0.004415431 0.004% explained by habitat

summary(eigenvals(test.rda, model = "constrained"))
#Importance of components:
#                       RDA1
#Eigenvalue            27.37
#Proportion Explained   1.00
#Cumulative Proportion  1.00

screeplot(test.rda)

signif.full <- anova.cca(test.rda, parallel=getOption("mc.cores")) # default is permutation=999
signif.full
#Permutation test for rda under reduced model
#Permutation: free
#Number of permutations: 999

#Model: rda(formula = genotypes ~ hab + Condition(Axis1 + Axis2 + Axis3), data = gen.imputed.dataset, scale = T)
#          Df Variance      F Pr(>F)    
#Model      1    27.37 1.5989  0.001 ***
#Residual 103  1763.14                  
#---
#Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

signif.axis <- anova.cca(test.rda, by="axis", parallel=getOption("mc.cores")) # default is permutation=999
signif.axis
#Permutation test for rda under reduced model
#Forward tests for axes
#Permutation: free
#Number of permutations: 999

#Model: rda(formula = genotypes ~ Habitat + Condition(Axis1 + Axis2 + Axis3), data = gen.imputed.dataset, scale = T)
#          Df Variance      F Pr(>F)    
#RDA1       1    27.37 1.5989  0.001 ***
#Residual 103  1763.14 

vif.cca(test.rda)
# Axis1           Axis2           Axis3 HabitatSylvatic 
#       1.002847        1.047709        1.109928        1.160484 

#Plot RDA results
load.rda.pop <- vegan::scores(test.rda, display = "site") 
pop.RDA <- as_tibble(cbind(load.rda.pop,phen))
load.rda <- vegan::scores(test.rda, display = "species") 
load.rda <- as_tibble(load.rda)

cols <- c("royalblue3", "chartreuse3")

p1 <- ggplot() +
  geom_point(data = load.rda, aes(x=RDA1, y=PC1),fill = "gray40", size = 1, shape = 21) +
  geom_point(data = pop.RDA, aes(x=RDA1, y=PC1,fill=Habitat),colour = "gray30", size = 4, shape = 21) +
  scale_fill_manual(values=c("royalblue3", "chartreuse3")) +
  geom_segment(aes(xend=test.rda$CCA$biplot[,1], yend=test.rda$CCA$biplot[,1], x=0, y=0),colour="black", size=2,linetype=1,arrow=arrow(length = unit(0.03, "npc"))) +
  geom_text(aes(x=1.8*test.rda$CCA$biplot[,1], y=1.8*test.rda$CCA$biplot[,1],
                label = "Wild ecotope"), size=10, colour = "black") + 
  xlab("RDA 1") + ylab("PC1") +
  theme_bw() +
   theme(axis.text.x = element_text( size=26,face="bold"), 
         axis.text.y = element_text(size=26,face="bold")) + 
   theme(axis.title.x = element_text(size=32, face="bold"),
         axis.title.y = element_text( size=32, face="bold")) +
  theme(legend.position = "none") 



density.inset<- ggplot() + 
  geom_density(data = pop.RDA, aes(x=RDA1,fill=Habitat), alpha = 0.75) +
  scale_fill_manual(values=c("royalblue3", "chartreuse3")) +
  theme(legend.position = "none") + 
  xlab("RDA 1") + ylab("Density") +
  theme_bw() +
   theme(axis.text.x = element_text( size=22,face="bold"), 
         axis.text.y = element_text(size=22,face="bold")) + 
   theme(axis.title.x = element_text(size=28, face="bold"),
         axis.title.y = element_text( size=28, face="bold")) +
  theme(legend.position = "none") 

png("RDA_SAM_V150421.png", units="in", width = 18, height = 10, res = 360)

p1 + annotation_custom(ggplotGrob(density.inset), 
                     xmin= 3, xmax=7, 
                       ymin = 1.5, ymax = 5.7)

dev.off()

ggplot() +
  geom_point(data = load.rda, aes(x=RDA1, y=PC1),fill = "gray40", size = 1, shape = 21) +
  geom_segment(aes(xend=test.rda$CCA$biplot[,1]/10, yend=test.rda$CCA$biplot[,1]/10, x=0, y=0),colour="red", size=0.5,linetype=1,arrow=arrow(length = unit(0.02, "npc"))) +
  geom_text(aes(x=1.2*test.rda$CCA$biplot[,1]/10, y=1.2*test.rda$CCA$biplot[,1]/10,
                label = colnames(phen[,2])), colour = "red") + 
  xlab("RDA 1") + ylab("PC1") 



#Z-transform snp loadings and identify outlier snps:
# This is the same data set used in random forest analysis
data <- read_csv("C:/Users/quiqu/Dropbox/Recuadoriensis_Analysis/RDA_analysis/RD_V2_analysis/gen.resid.dataset.csv", col_names = F)
data <- data[,6:2557]
snpid <- (t(data[1,]))

load.rda$zRDA1 <- ((load.rda$RDA1 - (mean(load.rda$RDA1)))/sd(load.rda$RDA1))

load.hist <- ggplot() +
  geom_histogram(data = load.rda, aes(x=RDA1), colour = "black", fill = "firebrick", alpha = 0.5, bins = 50) +
  xlab("loadings")

zload.hist <- ggplot() +
  geom_histogram(data = load.rda, aes(x=zRDA1), colour = "black", fill = "firebrick", alpha = 0.5, bins = 50) +
  xlab("z-transformed loadings")

loadings.plots <- plot_grid(load.hist, zload.hist, align = "h", labels = c("A", "B"), ncol = 2)

#
load.rda.id <- as_tibble(cbind(snpid, load.rda))

# Identify SNPs realted to habitat type. Here we assume negative RDA axis values are domicile
load.rda.id$habitat <- ifelse(load.rda.id$RDA1 < 0, "domicile", "sylvatic")


#Identify topsubset:
rda1.top2 <- subset(load.rda.id, abs(zRDA1) >= 2)
rda1.top3 <- subset(load.rda.id, abs(zRDA1) >= 3)



# SNPs at the tails of the distribution blue 2 standard deviation, red 3 Standard deviations
ggplot() +
  geom_point(data = load.rda.id, aes(x=RDA1, y=PC1),fill = "gray30", size = 1, shape = 21) +
  geom_point(data = rda1.top2, aes(x=RDA1, y=PC1),fill = "royalblue", size = 4, shape = 21) +
  geom_point(data = rda1.top3, aes(x=RDA1, y=PC1),fill = "firebrick", size = 4, shape = 21) +
  geom_segment(aes(xend=test.rda$CCA$biplot[,1]/10, yend=test.rda$CCA$biplot[,1]/10, x=0, y=0),colour="navyblue", size=0.5,linetype=1,arrow=arrow(length = unit(0.02, "npc"))) +
  geom_text(aes(x=1.2*test.rda$CCA$biplot[,1]/10, y=1.2*test.rda$CCA$biplot[,1]/10,
                label = colnames(phen[,2])), colour = "navyblue") + 
  xlab("RDA 1") + ylab("PC1")



##############################


rda.rf.top2 <- read_csv("C:/Users/quiqu/Dropbox/Recuadoriensis_Analysis/RDA_analysis/RD_V2_analysis/rda_top2_RF_accuracy.csv", col_names = T)

rda.rf.top3 <- read_csv("C:/Users/quiqu/Dropbox/Recuadoriensis_Analysis/RDA_analysis/RD_V2_analysis/rda_top3_RF_accuracy.csv", col_names = T)

strong_Snp <- read_csv("./rda_snp_15732.csv", col_names = T)

rda.rf.top2 <- read_csv("./rda_top2_RF_accuracy.csv", col_names = T)

rda.rf.top3 <- read_csv("./rda_top3_RF_accuracy.csv", col_names = T)

strong_Snp <- read_csv("./rda_snp_15732.csv", col_names = T)

png("RDA_SNPS_V150421.png", units="in", width = 18, height = 10, res = 360)

ggplot() +
  geom_point(data = load.rda.id, aes(x=RDA1, y=PC1), fill = "gray60", alpha=0.5, size = 1, shape = 21) +
  geom_point(data = rda1.top2, aes(x=RDA1, y=PC1),fill = "gray60", alpha=0.9, size = 4, shape = 21) +
  geom_point(data = rda.rf.top2, aes(x=RDA1, y=PC1),fill = "black", size = 4, shape = 21) +
  geom_point(data = strong_Snp, aes(x=RDA1, y=PC1),fill = "purple", size = 5, shape = 23) +
  geom_segment(aes(xend=test.rda$CCA$biplot[,1]/10, yend=test.rda$CCA$biplot[,1]/10, x=0, y=0),colour="black", size=2,linetype=1,arrow=arrow(length = unit(0.02, "npc"))) +
  geom_text(aes(x=1.2*test.rda$CCA$biplot[,1]/10, y=1.2*test.rda$CCA$biplot[,1]/10,
                label = "Wild ecotope"), size = 10, colour = "black") + 
  xlab("RDA 1") + ylab("PC1") +
  theme_bw() +
   theme(axis.text.x = element_text( size=26,face="bold"), 
         axis.text.y = element_text(size=26,face="bold")) + 
   theme(axis.title.x = element_text(size=32, face="bold"),
         axis.title.y = element_text( size=32, face="bold")) +
  theme(legend.position = "none") 

dev.off()


ggplot() +
  geom_point(data = load.rda.id, aes(x=RDA1, y=PC1),fill = "gray60", size = 1, shape = 21) +
  geom_point(data = rda1.top2, aes(x=RDA1, y=PC1),fill = "royalblue", size = 4, shape = 19) +
  geom_point(data = rda.rf.top2, aes(x=RDA1, y=PC1),fill = "purple", size = 4, shape = 21) +
  geom_segment(aes(xend=test.rda$CCA$biplot[,1]/10, yend=test.rda$CCA$biplot[,1]/10, x=0, y=0),colour="navyblue", size=0.5,linetype=1,arrow=arrow(length = unit(0.02, "npc"))) +
  geom_text(aes(x=1.2*test.rda$CCA$biplot[,1]/10, y=1.2*test.rda$CCA$biplot[,1]/10,
                label = colnames(phen[,2])), colour = "navyblue") + 
  xlab("RDA 1") + ylab("PC1") +
  theme_bw() +
   theme(axis.text.x = element_text( size=18,face="bold"), 
         axis.text.y = element_text(size=18,face="bold")) + 
   theme(axis.title.x = element_text(size=25, face="bold"),
         axis.title.y = element_text( size=25, face="bold")) +
  theme(legend.position = "none") 





write_csv(rda1.top2, "C:/Users/quiqu/Dropbox/Recuadoriensis_Analysis/RDA_analysis/RD_V2_analysis/rda_top2.csv")
write_csv(rda1.top3, "C:/Users/quiqu/Dropbox/Recuadoriensis_Analysis/RDA_analysis/RD_V2_analysis/rda_top3.csv")
write_csv(load.rda.id, "C:/Users/quiqu/Dropbox/Recuadoriensis_Analysis/RDA_analysis/RD_V2_analysis/load.rda.id.csv")

```
